### PR TITLE
fix: appbar header ignores border bottom radius

### PR DESCRIPTION
**Pull request by josemak25**
### Summary
This fix Implements: https://github.com/callstack/react-native-paper/issues/3569

- Add fix for app bar header custom styles
   -  This change addresses styles like border bottom radius on the app bar

### Test plan

1. Start the example app 
2. Click on the Appbar
3. Enable custom color switch
4. Enable Elevated
5. And lastly pass your custom styles to the app bar header

    ```javascript
    <Appbar.Header elevated={true} style={{ borderBottomLeftRadius: 16, borderBottomRightRadius: 16 }}>
       <Appbar.Content title="My Title" />
    </Appbar.Header>
    ```

<details>
<summary>screenshot from web</summary>

Screenshot 2023-03-20 at 2 12 14 PM

</details>

<details>
<summary>screenshot from web</summary>

mobile screenshot

</details>

### CODE REVIEW

**Review by DimitarNestorov — APPROVED** [review]

File: src/components/Appbar/AppbarHeader.tsx [hunk 1]
```diff
@@ -153,10 +155,12 @@ const AppbarHeader = ({
           paddingTop: statusBarHeight ?? top,
           paddingHorizontal: Math.max(left, right),
         },
+        restStyle,
```
Comment (lukewalczak) [inline src/components/Appbar/AppbarHeader.tsx:158]: I'm not sure if it's the correct approach to pass `restStyle` to the `Appbar` root layer (as you named it) which in styles has previously extracted styles properties from destructured style object 🤔 Maybe we should rather just pass the `borderRadius` into that layer?

File: src/components/Appbar/utils.ts [hunk 2]
```diff
@@ -39,6 +39,44 @@ export const getAppbarColor = (
...
+
+  const borders = Object.entries(properties)
+    .filter(([_, value]) => value)
+    .reduce<Record<string, typeof borderRadius>>((acc, [key, value]) => {
+      acc[key] = value;
+      return acc;
+    }, {});
+
+  return borders;
+};
```
Comment (lukewalczak) [inline src/components/Appbar/utils.ts:78]: Maybe we can simplify it as follows:

```
const borders = {};
for (const [key, value] of Object.entries(properties)) {
  if (value) {
    borders[key] = value;
  }
}

return borders;
```

wdyt @josemak25 @DimitarNestorov?

Comment (josemak25) [reply src/components/Appbar/utils.ts:78]: I agree this is much simpler to read, thanks for the suggestion 👍

Comment (josemak25) [reply src/components/Appbar/utils.ts:79]: Yea, this simplifies it even more

**Review by DimitarNestorov — APPROVED** [review]

**Pull request merged**